src/refactoring/components/admin/productManage/ProductInfo.tsx

```diff
@@ -0,0 +1,75 @@
+import { ChangeEvent } from 'react';
+
+import { excludeTargetIndexDiscount, getTargetProduct } from '@/refactoring/helpers';
+import { useEditingProduct, useNewDiscount, useOpenProductIds } from '@/refactoring/hooks';
+import { useProductStore } from '@/refactoring/stores';
+import { Product } from '@/types';
+
+import { ProductInfoDetail, ProductInfoEdit, ProductInfoSummary } from '..';
+
+interface Props {
+  product: Product;
+  index: number;
+}
+
+export const ProductInfo = ({ product, index }: Props) => {
+  const products = useProductStore((state) => state.products);
+  const updateProduct = useProductStore((state) => state.updateProduct);
+
+  const { openProductIds, toggleProductAccordion } = useOpenProductIds();
+  const { editingProduct, updateEditingProduct, handleEditingProductUpdate } = useEditingProduct();
+  const { newDiscount, addDiscount, handleUpdateDiscount } = useNewDiscount();
+
+  const handleEditComplete = () => {
```

Comment:
이렇게 수정하면 `handleEditComplete`를 선언하는 대신
조금 더 선언적인 모습으로 바꿀 수 있을 것 같습니다!

`editingProduct`를 초기화하는 로직은 훅 내부에 숨기고, 컴포넌트는 수정된 product를 이용해서 `updateProduct`를 호출하는 로직에만 집중할 수 있을 것 같아요 :-)

```suggestion
  const { editingProduct, updateEditingProduct, handleEditingProductUpdate } = useEditingProduct({
    onEditComplete: (product) => {
      updateProduct(product);
    }
  });
```

[비즈니스 로직 분리에 대한 글](https://blog.leehov.in/57)

이 글을 읽고 비즈니스 로직 분리에 대해 살짝 감이 왔는데, 한번 읽어보셔도 좋을 것 같습니다 !!